modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -2597,8 +2600,28 @@ private GridTuple3<Long, Long, Boolean> ttlAndExpireTime(IgniteCacheExpiryPolicy
      */
     private void drReplicate(GridDrType drType, @Nullable CacheObject val, GridCacheVersion ver, AffinityTopologyVersion topVer)
         throws IgniteCheckedException {
-        if (cctx.isDrEnabled() && drType != DR_NONE && !isInternal())
+        if (cctx.isDrEnabled() && drType != DR_NONE && !isInternal()) {
+            traceKeyReplication(ver);
+
             cctx.dr().replicate(key, val, rawTtl(), rawExpireTime(), ver.conflictVersion(), drType, topVer);
+        }
+    }
+
+    private void traceKeyReplication(GridCacheVersion ver) {
+        if (log.isTraceEnabled()) {
+            GridToStringBuilder.SensitiveDataLogging sensitiveDataLogging = S.getSensitiveDataLogging();
+            String keyValue = "nil";
+
+            if (sensitiveDataLogging == PLAIN)
+                keyValue = keyValue(false).toString();
+            else if (sensitiveDataLogging == HASH)
+                keyValue = String.valueOf(IgniteUtils.hash(keyValue(false)));
```

Comment:
Let's move this code to a separate method `keyValueToLog()` and use it below instead of `keyValue(false)`